pkg/hintrunner/zero/zerohint_ec.go

```diff
@@ -0,0 +1,97 @@
+package zero
+
+import (
+	"fmt"
+	"github.com/NethermindEth/cairo-vm-go/pkg/hintrunner/hinter"
+	"github.com/NethermindEth/cairo-vm-go/pkg/utils"
+	VM "github.com/NethermindEth/cairo-vm-go/pkg/vm"
+	"github.com/consensys/gnark-crypto/ecc/stark-curve/fp"
+	"math/big"
+)
+
+func newEcNegateHint(point hinter.ResOperander) hinter.Hinter {
+	return &GenericZeroHinter{
+		Name: "EcNegate",
+		Op: func(vm *VM.VirtualMachine, ctx *hinter.HintRunnerContext) error {
+			//> from starkware.cairo.common.cairo_secp.secp_utils import SECP_P, pack
+			//> y = pack(ids.point.y, PRIME) % SECP_P
+			//> # The modulo operation in python always returns a nonnegative number.
+			//> value = (-y) % SECP_P
+
+			secPBig, ok := utils.GetSecPBig()
+			if !ok {
+				return fmt.Errorf("GetSecPBig failed")
+			}
+
+			baseBig, ok := utils.GetEcBaseBig()
+			if !ok {
+				return fmt.Errorf("GetEcBaseBig failed")
+			}
+
+			pointValues, err := hinter.GetConsecutiveValues(vm, point, int16(6))
+			if err != nil {
+				return err
+			}
+
+			yD0, err := pointValues[3].FieldElement()
```

Comment:
Maybe keep the generation of array in the hint and move the pack function to the separate utility?

Reply:
I've moved some functionality to reusable util methods.